fedml_core/trainer/model_trainer.py

```diff
@@ -24,9 +26,11 @@ def set_model_params(self, model_parameters):
         pass
 
     @abstractmethod
-    def train(self, train_data, device, args):
+    def train(self, train_data, device, args=None):
```

Comment:
Once we set the args as a member variable why do we still need to pass it in the class methods?

Reply:
Setting it as a member variable is something I am proposing to do. Before this commit, it was just passed as argument to two specific functions. However, there are other projects using these core trainer files and if I remove args right away then their code will break. So, we need to remove it after they have made the changes as well. Have added _TODO comments_ mentioning the same. Also, I have mentioned the same reasoning in bit more detailed manner in the PR conversation itself incase you missed it.

Reply:
Yeah, I saw that. Got it